### DISCUSSION TITLE
Is it possible to have vulnerabilityAlerts only create MRs for minor and patch versions?

### DISCUSSION THREAD

**Discussion by Wayneoween** [renovatebot/renovate#32659] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I'm trying to find out if it is possible to limit `vulnerabilityAlerts` to `minor` and `patch` and exclude `major` versions.

E.g. currently `renovate` creates MRs like this, which wants to upgrade `vue` to a major version but this is already on our radar mid-term.

![image](https://github.com/user-attachments/assets/ba14f28b-e680-41c8-bfb9-33b7e0ac52b7)

We have set up `dependencyDashboardApproval` for `major` versions, but `vulnerabilityAlerts` seems to ignore this (fair enough, but that's why I'm here). Is it even possible to configure this for `vulnerabilityAlerts` or is it opinionated in that way, that this can't be done?

If it possible what would a minimal example look like? I couldn't figure out something by looking at the documentation or [this discussion](https://github.com/renovatebot/renovate/discussions/20542).

I imagined something like this:

```json
{
  "packageRules": [
    {
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "vulnerabilityAlerts": {
        "enabled": true,
        "labels": [
          "type:renovate"
        ]
      }
    },
  ],
  "osvVulnerabilityAlerts": true
}
```

Or the other way around, but I didn't manage to get it to work so I'm trying to find help here because it might not even be possible.

**Comment by RahulGautamSingh** [comment, marked as answer]
We don't want that to happen as users could miss the VAs that way. But, maybe some config is possible to get that result as well, you'll have to hit & try.

One way is to disable major updates for all deps, that would affect the VAs too..using the preset `:disableMajorUpdates`

**Reply by Wayneoween** [reply]
Thank you, in my test repository `:disableMajorUpdates` seems to work.

I fully understand that this is a weird question and that this feature shouldn't be limited to a certain type of update.

We'll have to adapt our workflow and may just use `dependencyDashboardApproval ` or simply close these kinds of MRs. But I will have to ask the team how we want to go forward.

Thank you for the prompt response @RahulGautamSingh
